YubiKit/YubiKit/Helpers/YKFTLVRecord.m

```diff
@@ -0,0 +1,155 @@
+// Copyright 2018-2022 Yubico AB
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#import <Foundation/Foundation.h>
+#import "YKFTLVRecord.h"
+#import "YKFNSDataAdditions+Private.h"
+
+@interface YKFTLVRecord()
+@property (nonatomic, readwrite) YKFTLVTag tag;
+@property (nonatomic, readwrite) NSData *value;
+@end
+
+@implementation YKFTLVRecord
+
++ (nullable instancetype)recordFromData:(NSData *_Nullable)data checkMatchingLength:(Boolean)checkMatchingLength bytesRead:(int*)bytesRead {
+    // tag
+    if (data.length == 0) { return nil; }
+    Byte *bytes = (Byte *)data.bytes;
+    int dataStartLocation = 2;
+    int tag = bytes[0] & 0xFF;
+    if ((tag & 0x1F) == 0x1F) {
+        tag = (tag << 8) | (bytes[1] & 0xFF);
+        dataStartLocation++;
+        int index = 1;
+        while ((tag & 0x80) == 0x80 && index < data.length) {
+            tag = (tag << 8) | (bytes[index] & 0xFF);
```

Comment:
Since `bytes[1]` was pulled in above, should `bytes[1]`  be pulled in here again?  (index is set to 1 on the first pass)

Reply:
It's already fixed. Think you accidentally commented on an old change set.